packages/core/src/resolve-namespace-factories.ts

```diff
@@ -10,25 +10,43 @@ export function packageNamespaceFactory(
     }: { dirname(path: string): string; relative(from: string, to: string): string },
     hashSalt = '',
     prefix = '',
-    normalizeVersion = (semver: string) => semver
+    normalizeVersion = (semver: string, _packageName: string) => semver,
+    buildNamespace = defaultBuildNamespace
 ): typeof processNamespace {
     return (namespace: string, stylesheetPath: string) => {
         const configPath = findConfig('package.json', { cwd: dirname(stylesheetPath) });
         if (!configPath) {
             throw new Error(`Could not find package.json for ${stylesheetPath}`);
         }
         const config = loadConfig(configPath) as { name: string; version: string };
-        const fromRoot = relative(dirname(configPath), stylesheetPath).replace(/\\/g, '/');
-        return (
-            prefix +
-            namespace +
-            murmurhash3_32_gc(
-                hashSalt + config.name + '@' + normalizeVersion(config.version) + '/' + fromRoot
-            )
+        return buildNamespace(
+            hashSalt,
```

Comment:
`hashSalt` value should be dependent on `config` in line `21` otherwise the `namespace` will be still linked to the artifactId (see the change I had to do [here](https://github.com/johnbenz13/mf-with-css-encapsulation/pull/3/files#diff-d1ccad9baa23796c3419806748865da9241ed48b4c3983945c1a1c6eb294ac9cR53))
so we could make it a function as well, something like:

```
hashSalt: (config): (config: {name: string; version: string}) => string
```

Reply:
The buildNamespace function can ignore this salt same way you did base on the package name so this still enable the same transformation. Also you will not use the normalizeVersion function and the entire logic can sit in the same place